### PR TITLE
Implementation of Bus Ticket Validator

### PR DESCRIPTION
**Pull request by PaterIntellectus**

### CODE REVIEW

File: src/main/java/com/andersen/BusTicketValidator/BusTicket.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    String ticketType;
+
+    String startDate;
+
+    int price;
+
+
+    public BusTicket() {
+        this.ticketClass = null;
```
Comment (deolexx) [inline src/main/java/com/andersen/BusTicketValidator/BusTicket.java:15]: could be shortened to just public BusTicket(){}

File: src/main/java/com/andersen/BusTicketValidator/BusTicket.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
+package com.andersen.BusTicketValidator;
+
+public class BusTicket {
+
+    String ticketClass;
```
Comment (deolexx) [inline src/main/java/com/andersen/BusTicketValidator/BusTicket.java:5]: private modifiers missing

File: src/main/java/com/andersen/BusTicketValidator/TicketValidator.java [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+
+enum ViolationType {
+    None, InvalidTicketType,
+    MissingStartDate, FutureStartDate, WrongStartDateFormat,
+    PriceIsZero, PriceIsOdd,
+}
+
+public class TicketValidator {
+    private static final HashSet<String> validTicketTypes = new HashSet<>();
+    static {
```
Comment (deolexx) [inline src/main/java/com/andersen/BusTicketValidator/TicketValidator.java:22]: just use validTicketTypes = Set.of("DAY","WEEK","MONTH","YEAR") at line 21

**Review by deolexx — APPROVED** [review]
LGTM

File: src/main/java/com/andersen/BusTicketValidator/BusTicket.java (hunk 1 above)
Comment (PaterIntellectus) [reply src/main/java/com/andersen/BusTicketValidator/BusTicket.java:15]: Or be deleted at all :P
I don't know why I left this default constructor at all

File: src/main/java/com/andersen/BusTicketValidator/TicketValidator.java (hunk 3 above)
Comment (PaterIntellectus) [reply src/main/java/com/andersen/BusTicketValidator/TicketValidator.java:22]: Good to know, when I was trying to find the way to initialize Set in Java Class, I couldn't find other way to do it other than the static initialization

**Pull request merged**